common/src/state/pending_message.rs

```diff
@@ -29,20 +29,33 @@ impl PendingMessage {
         PendingMessage {
             attachments: attachments
                 .iter()
-                .map(|p| {
-                    if let Some(name) = p.file_name().map(|ostr| ostr.to_str().unwrap_or_default())
-                    {
-                        return name.to_string();
+                .map(|p| match p {
```

Comment:
Maybe this should be a filter_map so we can filter and do something like

```rust
let path = match p {
   Location::Disk { path } => path,
   Location::Constellation { path } => PathBuf::from(path),
};

path.file_name().and_then(OsStr::to_str).map(str::to_string)
```

This would simplify the code and if `PathBuf::file_name` returned `None` it would be filtered out.